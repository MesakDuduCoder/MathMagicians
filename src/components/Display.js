import React from 'react';
import PropTypes from 'prop-types';

function Display({ total, operation, next }) {
  console.log(typeof (total));
  // const next = props.display.next || '';
  // const operation = props.display.operation || '';
  return (
    <div className="result">
      {total}
      {' '}
      {operation}
      {' '}
      {next}
    </div>
  );
}

Display.propTypes = {
  total: PropTypes.string.isRequired,
  operation: PropTypes.string.isRequired,
  next: PropTypes.string.isRequired,
};

export default Display;
