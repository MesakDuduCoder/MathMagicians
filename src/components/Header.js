import React from 'react';
import { Link } from 'react-router-dom';
import '../css/header.css';

function Header() {
  return (
    <div className="header">
      <h1>Math Magicians</h1>
      <nav>
        <Link className="links" to="/math-magicians">Home</Link>
        <Link className="links center" to="/calculator">Calculator</Link>
        <Link className="links" to="/quote">Quote</Link>
      </nav>
    </div>
  );
}

export default Header;
