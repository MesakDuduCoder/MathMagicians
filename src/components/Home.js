import React from 'react';

function Home() {
  return (
    <div>
      <h2>Welcome to our page!</h2>
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Minus
        praesentium temporibus in voluptatum voluptate animi mollitia fugiat
        esse, culpa ex exercitationem possimus expedita vero beatae similique
        repellat debitis blanditiis labore. Tenetur error dolore necessitatibus
        dolorem enim quasi tempora eos veritatis, eius ex dignissimos
      </p>
      <p>
        Laboriosam aperiam excepturi cupiditate, eligendi voluptatem ad! In odit
        incidunt officia. Pariatur sed dolorem aliquam vel omnis, facere
        quibusdam quasi, quos minima sequi saepe? Libero voluptas enim aperiam
        tenetur saepe? Molestiae culpa, qui illo quo asperiores, provident
        officia magnam illum eius consequatur labore, dolore unde eveniet?
        Reiciendis omnis perspiciatis, voluptate totam nihil, veniam corrupti
        vitae inventore minus voluptatum, neque nisi!
      </p>
    </div>
  );
}

export default Home;
