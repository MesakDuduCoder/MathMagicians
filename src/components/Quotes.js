import React, { useEffect, useState } from 'react';
import '../css/quotes.css';

function Quotes() {
  const [quote, setQuote] = useState('');
  const [loading, setLoading] = useState(true);
  const [error, setError] = useState(null);

  useEffect(() => {
    const requestOptions = {
      method: 'GET',
      headers: {
        'Content-Type': 'application/json',
        'X-Api-Key': 'NN95259E13YZ0ovCcyfupA==HrGoapCZpfPyBdj6',
      },
    };

    async function fetchData() {
      try {
        setLoading(true);
        const response = await fetch(
          'https://api.api-ninjas.com/v1/quotes?category=architecture',
          requestOptions,
        );
        const data = await response.json();
        const newData = `${data[0].quote}\u00A0-\u00A0'${data[0].author}'`;
        setQuote(newData);
        setLoading(false);
      } catch (error) {
        setError(error);
      }
    }
    fetchData();
  }, []);

  return (
    <div className="quote">
      {error ? (
        <div className="error">
          <h1>Error</h1>
          <p>There was an error cannot show quote</p>
        </div>
      ) : (
        <div className="text">{loading ? <>Loading..</> : <>{quote}</>}</div>
      )}
    </div>
  );
}

export default Quotes;
