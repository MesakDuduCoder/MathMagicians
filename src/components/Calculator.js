import React, { useState } from 'react';
import Display from './Display';
import '../css/calculator.css';
import calculate from '../logic/calculate';

function Calculator() {
  const [state, setState] = useState(() => ({
    total: null,
    next: null,
    operation: null,
  }));

  function submit(e, number) {
    e.preventDefault();
    setState((prevState) => calculate(prevState, number));
  }
  const total = state.total || '';
  const operation = state.operation || '';
  const next = state.next || '';

  return (
    <div className="container">
      <div className="result" data-testid="display">
        <Display total={total} operation={operation} next={next} />
      </div>
      <div className="numbers-container">
        <div className="numbers-row">
          <button type="button" onClick={(e) => submit(e, "AC")}>
            AC
          </button>
          <button type="button" onClick={(e) => submit(e, "+/-")}>
            +/-
          </button>
          <button type="button" onClick={(e) => submit(e, "%")}>
            %
          </button>
          <button
            className="operator-btn"
            type="button"
            onClick={(e) => submit(e, "รท")}
          >
            รท
          </button>
        </div>
        <div className="numbers-row">
          <button data-testid="7" type="button" onClick={(e) => submit(e, "7")}>
            7
          </button>
          <button type="button" onClick={(e) => submit(e, "8")}>
            8
          </button>
          <button type="button" onClick={(e) => submit(e, "9")}>
            9
          </button>
          <button
            className="operator-btn"
            type="button"
            onClick={(e) => submit(e, "x")}
          >
            x
          </button>
        </div>
        <div className="numbers-row">
          <button type="button" onClick={(e) => submit(e, "4")}>
            4
          </button>
          <button type="button" onClick={(e) => submit(e, "5")}>
            5
          </button>
          <button type="button" onClick={(e) => submit(e, "6")}>
            6
          </button>
          <button
            className="operator-btn"
            type="button"
            onClick={(e) => submit(e, "-")}
          >
            -
          </button>
        </div>
        <div className="numbers-row">
          <button type="button" onClick={(e) => submit(e, "1")}>
            1
          </button>
          <button type="button" onClick={(e) => submit(e, "2")}>
            2
          </button>
          <button type="button" onClick={(e) => submit(e, "3")}>
            3
          </button>
          <button
            className="operator-btn"
            type="button"
            onClick={(e) => submit(e, "+")}
          >
            +
          </button>
        </div>
        <div className="numbers-row">
          <button
            className="zero"
            type="button"
            onClick={(e) => submit(e, "0")}
          >
            0
          </button>
          <button type="button" onClick={(e) => submit(e, ".")}>
            .
          </button>
          <button
            className="operator-btn"
            type="button"
            onClick={(e) => submit(e, "=")}
          >
            =
          </button>
        </div>
      </div>
    </div>
  );
}

export default Calculator;
