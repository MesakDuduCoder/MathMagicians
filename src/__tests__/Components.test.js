import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Home from '../components/Home';
import Header from "../components/Header";
import Calculator from "../components/Calculator";
import renderer from "react-test-renderer";
import Quotes from "../components/Quotes";
import Display from "../components/Display";
import { MemoryRouter } from "react-router-dom";

describe('Operations should be as below', () => {
    it("renders correctly", () => {
      const calulator = renderer
        .create(<Calculator/>)
        .toJSON();
      expect(calulator).toMatchSnapshot();
    });

    it("renders correctly", () => {
      const quotes = renderer
        .create(<Quotes/>)
        .toJSON();
      expect(quotes).toMatchSnapshot();
    });

    it("renders correctly", () => {
      const display = renderer
        .create(<Display/>)
        .toJSON();
      expect(display).toMatchSnapshot();
    });

    it("renders correctly", () => {
      const home = renderer
        .create(<Home/>)
        .toJSON();
      expect(home).toMatchSnapshot();
    });
    
    it("renders correctly", () => {
      const header = renderer
        .create(
          <MemoryRouter>
            <Header />
          </MemoryRouter>
        )
        .toJSON();
      expect(header).toMatchSnapshot();
    });

    it("will test user interaction", () => {
      render(<Calculator/>);

      userEvent.click(screen.getByTestId("7"));
    
      expect(screen.getByTestId("display")).toHaveTextContent("7");

    });
})