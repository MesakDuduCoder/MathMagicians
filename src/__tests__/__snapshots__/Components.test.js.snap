// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Operations should be as below renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="result"
    data-testid="display"
  >
    <div
      className="result"
    >
       
       
    </div>
  </div>
  <div
    className="numbers-container"
  >
    <div
      className="numbers-row"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="operator-btn"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
    </div>
    <div
      className="numbers-row"
    >
      <button
        data-testid="7"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="operator-btn"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
    </div>
    <div
      className="numbers-row"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="operator-btn"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
    </div>
    <div
      className="numbers-row"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="operator-btn"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </div>
    <div
      className="numbers-row"
    >
      <button
        className="zero"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="operator-btn"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;

exports[`Operations should be as below renders correctly 2`] = `
<div
  className="quote"
>
  <div
    className="text"
  >
    Loading..
  </div>
</div>
`;

exports[`Operations should be as below renders correctly 3`] = `
<div
  className="result"
>
   
   
</div>
`;

exports[`Operations should be as below renders correctly 4`] = `
<div>
  <h2>
    Welcome to our page!
  </h2>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Minus praesentium temporibus in voluptatum voluptate animi mollitia fugiat esse, culpa ex exercitationem possimus expedita vero beatae similique repellat debitis blanditiis labore. Tenetur error dolore necessitatibus dolorem enim quasi tempora eos veritatis, eius ex dignissimos
  </p>
  <p>
    Laboriosam aperiam excepturi cupiditate, eligendi voluptatem ad! In odit incidunt officia. Pariatur sed dolorem aliquam vel omnis, facere quibusdam quasi, quos minima sequi saepe? Libero voluptas enim aperiam tenetur saepe? Molestiae culpa, qui illo quo asperiores, provident officia magnam illum eius consequatur labore, dolore unde eveniet? Reiciendis omnis perspiciatis, voluptate totam nihil, veniam corrupti vitae inventore minus voluptatum, neque nisi!
  </p>
</div>
`;

exports[`Operations should be as below renders correctly 5`] = `
<div
  className="header"
>
  <h1>
    Math Magicians
  </h1>
  <nav>
    <a
      className="links"
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
    <a
      className="links center"
      href="/calculator"
      onClick={[Function]}
    >
      Calculator
    </a>
    <a
      className="links"
      href="/quote"
      onClick={[Function]}
    >
      Quote
    </a>
  </nav>
</div>
`;
